src/s1reader/s1_orbit.py

```diff
@@ -204,35 +218,43 @@ def get_file_name_tokens(zip_path: str) -> [str, list[datetime.datetime]]:
 # lambda to check if file exists if desired sat_id in basename
 item_valid = lambda item, sat_id: os.path.isfile(item) and sat_id in os.path.basename(item)
 
-
-def get_orbit_file_from_list(zip_path: str, orbit_file_list: list[str]) -> str:
+def get_orbit_file_from_dir(zip_path: str, orbit_dir: str, auto_download: bool = False) -> str:
     '''Get orbit state vector list for a given swath.
 
     Parameters:
     -----------
     zip_path : string
-        Path to Sentinel1 SAFE zip file. File names required to adhere to the
+        Path to Sentinel1 SAFE zip file. Base names required to adhere to the
         format described here:
         https://sentinel.esa.int/web/sentinel/user-guides/sentinel-1-sar/naming-conventions
-    orbit_file_list: list[str]
-        List containing orbit files paths. Orbit files required to adhere to
-        naming convention found here:
+    orbit_dir : string
+        Path to directory containing orbit files. Orbit files required to adhere
+        to naming convention found here:
         https://s1qc.asf.alaska.edu/aux_poeorb/
+    auto_download : bool
+        Automatically download the orbit file if not exist in the orbit_dir.
 
     Returns:
     --------
-    orbit_path : str
-        Path the orbit file.
+    orbit_file : str
+        Path to the orbit file.
     '''
+
+    # check the existence of input file path and directory
     if not os.path.exists(zip_path):
         raise FileNotFoundError(f"{zip_path} does not exist")
 
-    if not orbit_file_list:
-        raise ValueError("no orbit paths provided")
+    if not os.path.isdir(orbit_dir):
```

Comment:
Oh one thing I got while testing this- Should we just make the directory if we specify one that doesn't exist with a `os.makedirs(orbit_dir, exist_ok=True)`? I'm thinking of integrating this into Virginia's stack workflow, and it will make a new directory for orbits as long as you have one with the zip files https://github.com/opera-adt/COMPASS/pull/18

Reply:
Either is fine with me, I don't have a preference for this. 

I did not do that because 1) my orbit directory always exists (since the same one is used for all the processing); 2) occasionally I make typos on the path, thus, creating-dir-if-not-exist does not suit my need.

Reply:
ok that seems equally good to make sure they user has typed in the correct orbit directory 👍